lib/src/core_matchers.dart

```diff
@@ -614,9 +614,13 @@ class CustomMatcher extends Matcher {
   featureValueOf(actual) => actual;
 
   bool matches(item, Map matchState) {
-    var f = featureValueOf(item);
-    if (_matcher.matches(f, matchState)) return true;
-    addStateInfo(matchState, {'feature': f});
+    try {
+      var f = featureValueOf(item);
+      if (_matcher.matches(f, matchState)) return true;
+      addStateInfo(matchState, {'feature': f});
```

Comment:
Since `matchState` has a global namespace, it's probably a good idea to prefix these keys with the matcher name—`custom.feature` and so on.